src/utils.ts

```diff
@@ -173,6 +193,35 @@ export async function resolveLabel(label: string, config: Config): Promise<strin
   return config.provider.resolveName(`${label}.${config.registrar.domain}`);
 }
 
+export async function lookupAddress(address: string, did: string | null, config: Config): Promise<Lookup>  {
+  // First get eventual ENS Registration
+  let label: string | null = null;
+  try {
+    label = await config.provider.lookupAddress(address);
+  } catch (error) {
+    console.log("LookupAddress: Not able to get ENS Name, trying with IDX");
+  }
+  if (label && await config.provider.resolveName(label)) {
+    return {type: "ens", content: await getRegistration(label, config) as Registration };
+  }
+  // Else lookup IDX profile
+  if (did && isDid(did)) {
+    const content = await resolveIdxProfile(did, config);
+    if (content) return { type: "idx", content };
+  }
+  // If everything fails return address
+  return { type: "address", content: { name: address }};
+}
```

Comment:
I'm not sure how this function will be used, so can't really comment on whether it makes sense. But one issue I can see is that it returns either an IDX profile, or an ENS name, where it might be more useful to return both, ie. anything we can find, and let the frontend component decide what to display.

Reply:
Okay yeah that would be fine too.
To avoid overfetching I was thinking of not fetching everything all the time, just the necessary according to the defined heuristic.

Reply:
27e9a28f819fb09a9c4ea3796d96362b28f89658 solves this